pkg/ebpf/btf.go

```diff
@@ -210,24 +229,24 @@ func (b *orderedBTFLoader) checkForUnminimizedBTF(extractDir string) (*returnBTF
 		if err != nil {
 			return nil, err
 		}
-		return &returnBTF{vmlinux: spec, moduleLoadFunc: modLoadFunc}, nil
+		return &btfSpecData{sourceFilepath: sourceTarball, vmlinux: spec, moduleLoadFunc: modLoadFunc}, nil
 	}
 	return nil, nil
 }
 
-func (b *orderedBTFLoader) checkforBTF(extractDir string) (*returnBTF, error) {
-	ret, err := b.checkForMinimizedBTF(extractDir)
+func (b *orderedBTFLoader) checkforBTF(extractDir, sourceTarball string) (*btfSpecData, error) {
+	ret, err := b.checkForMinimizedBTF(extractDir, sourceTarball)
```

Comment:
I would prefer you add the path information to `ret` here, if not nil, rather than changing all these function signatures to just pass through the information.

Reply:
This applies to all the `load*` functions in this file.

Reply:
Updated this to store the data in `resultMetadata` which persists after the actual BTF data gets flushed